simul/platform/mininet/setup_servers.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash -e
+cd "$(dirname ${BASH_SOURCE[0]})"
+
+SERVER_GW="$1"
+SERVERS="$@"
+KEYS=/tmp/server_keys
+
+rm -f $KEYS
+for s in $SERVERS; do
+	echo Starting to install on $s
+	login=root@$s
+	ip=$( host $s | sed -e "s/.* //" )
+	ssh-keygen -R $s > /dev/null
+	ssh-keygen -R $ip  > /dev/null
+	ssh-keyscan -t ssh-ed25519 $s >> ~/.ssh/known_hosts 2> /dev/null
+	ssh-copy-id $login 2> /dev/null
+	ssh $login "test ! -f .ssh/id_rsa && echo -e '\n\n\n\n' | ssh-keygen > /dev/null" || true
+	ssh $login cat .ssh/id_rsa.pub >> $KEYS
+	if ! ssh $login "egrep -q '(14.04|Debian GNU/Linux 8)' /etc/issue"; then
+		clear
+		echo "$s does not have Ubuntu 14.04 or Debian 8 installed - aborting"
+		exit 1
+	fi
+	scp install_mininet.sh $login: > /dev/null
+	ssh -f $login "./install_mininet.sh &> /dev/null"
+done
+
+DONE=0
+NBR=$( echo $SERVERS | wc -w )
+while [ $DONE -lt $NBR ]; do
+	DONE=0
+	clear
+	echo "$( date ) - Waiting on $NBR servers - $DONE are done"
+	for s in $SERVERS; do
+		if ! ssh root@$s "ps ax | grep -v ps | grep install | grep -q mininet"; then
+			DONE=$(( DONE + 1 ))
+		else
+			echo -e "\nProcesses on $s"
+			ssh root@$s 'pstree -p $( ps ax | grep "bash ./install_mininet.sh" | grep -v grep | sed -e "s/ *\([0-9]*\) .*/\1/" )'
+		fi
+	done
+	sleep 2
+done
+
+echo -e "\nAll servers are done installing - copying ssh-keys"
+
+rm -f server_list
+for s in $SERVERS; do
+	login=root@$s
+	cat $KEYS | ssh $login "cat - >> .ssh/authorized_keys"
```

Comment:
I start to set up `ssh-copy-id` between the host-machine and the servers, and this will ask for a password for every connection. In order not to have that prompt for every connection between the servers, I copy the public-keys with this line. Saves (n^2 - n) password-entries.